### DISCUSSION TITLE
Do I have to release / delete / destroy windows somehow?

### DISCUSSION THREAD

**Discussion by yngwi** [fltk-rs/fltk-rs#1312] [Q&A] [answered]
I'm creating an app where multiple similar windows can be opened / closed at the same time (similar to editors for various files). The windows are part of custom structs that are stored in a hashmap. When a window is closed with the "x" button, the object containing the window is removed from the hashmap. Is this sufficient to permanently remove these windows or do I somehow also have to delete them with a fltk-rs native mechanism? Thank you for your help!

**Comment by MoAlyousef** [comment, marked as answer]
Usually FLTK assumes windows are long lived for the entirety of the program and it also assumes that if a window is closed, it might be opened again. For temporary windows it might be better to delete them.

**Reply by yngwi** [reply]
Thank you but how can I actually delete the window? I assume with `WidgetBase::delete()` but I don't understand how the call has to look. I'm sorry for this noob question.

**Reply by MoAlyousef** [reply]
It's a static method:
```rust
use fltk::{prelude::*, *};

fn main() {
    let app = app::App::default();
    let mut win = window::Window::new(100, 100, 800, 600, None);
    win.end();
    win.show();
    app.run().unwrap();
    window::Window::delete(win);
}

```

**Reply by yngwi** [reply]
Thank you
